txn_writer.go

```diff
@@ -316,13 +321,13 @@ func (sw *SafeWriter) Write(root string, sm gps.SourceManager, noExamples bool)
 
 		// 0666 is before umask; mirrors behavior of os.Create (used by
 		// writeFile())
-		if err = ioutil.WriteFile(filepath.Join(td, ManifestName), append([]byte(initOutput), tb...), 0666); err != nil {
+		if err := writeFile(filepath.Join(td, ManifestName), sw.Manifest, initOutput); err != nil {
```

Comment:
It doesn't look like `writeFile` sets the `0666` mask, is that important? Also, this is going to double-call MarshalTOML() on `sw.Manifest` -- any reason not to hang onto that value?